android/src/main/java/com/wenkesj/voice/VoiceModule.java

```diff
@@ -43,6 +44,26 @@ public String getName() {
 
   @ReactMethod
   public void startSpeech(final String locale, final Callback callback) {
+    if (!isPermissionGranted()) {
```

Comment:
I like this idea here it's very graceful, but I think still this should be a flag on the new options object passed with `Voice.start`. For instance:
```js
if (!isPermissionGranted() && options.getBool("REQUEST_PERMISSIONS_AUTO")) {
// ...
}
```
.